services/github/github-package-lock-json.service.js

```diff
@@ -0,0 +1,281 @@
+'use strict'
+
+const { renderVersionBadge } = require('../version')
+const {
+  transformAndValidate,
+  renderDynamicBadge,
+} = require('../dynamic-common')
+const {
+  isPackageLockJsonWithDependencies,
+  getLockDependencyVersion,
+} = require('../package-lock-json-helpers')
+const { semver } = require('../validators')
+const { ConditionalGithubAuthV3Service } = require('./github-auth-service')
+const { fetchJsonFromRepo } = require('./github-common-fetch')
```

Comment:
Also want to note that this using the GH contents v3 API that has a file limit of 1MB, which will be problematic in a lot of instances given the sizes of the lockfiles

https://github.com/badges/shields/issues/4114#issuecomment-538531437